crates/control/src/main.rs

```diff
@@ -36,10 +37,13 @@ async fn main() -> anyhow::Result<()> {
     let (put_builds, fetch_builds) = init_builds_root(&settings.builds_root)?;
     let ctx = AppContext::new(db, put_builds, fetch_builds);
 
-    let server = startup::run(listener, ctx)?;
+    let server = startup::run(listener, ctx.clone())?.map_err(Into::into);
+    let builder_daemon = control::services::builder::serve_builds(ctx.clone(), shutdown::signal())
+        .map_err(Into::into);
 
-    // The server runs until it receives a shutdown signal.
-    server.await?;
+    // Run until the builder_daemon and server both exit.
+    let out: Result<_, anyhow::Error> = futures::try_join!(server, builder_daemon);
+    out?;
```

Comment:
nit: can this be
```suggestion
    futures::try_join!(server, builder_daemon);
```

Reply:
1) We want to return a Result::Error
2) It needs the return value type hint to figure out how to Into::into the respective future error types.